From 457c32a9980c90255ae78690626c925238cd7485 Mon Sep 17 00:00:00 2001
From: Michael Goffioul <michael.goffioul@lincor.com>
Date: Fri, 4 Oct 2019 20:11:48 -0400
Subject: [PATCH 12/15] Disable CFI

The presence of CFI sanitizer makes the corresponding libraries to be
missing from the com.android.media.swcodec APEX, when building for x86.
---
 media/codec2/components/base/Android.bp    | 3 ---
 media/codec2/sfplugin/utils/Android.bp     | 1 -
 media/libstagefright/bqhelper/Android.bp   | 1 -
 media/libstagefright/flac/dec/Android.bp   | 1 -
 media/libstagefright/foundation/Android.bp | 1 -
 5 files changed, 7 deletions(-)

diff --git a/media/codec2/components/base/Android.bp b/media/codec2/components/base/Android.bp
index 160e250235..87d49a3a92 100644
--- a/media/codec2/components/base/Android.bp
+++ b/media/codec2/components/base/Android.bp
@@ -39,7 +39,6 @@ cc_library {
             "unsigned-integer-overflow",
             "signed-integer-overflow",
         ],
-        cfi: true,
     },
 
     ldflags: ["-Wl,-Bsymbolic"],
@@ -87,7 +86,6 @@ cc_defaults {
             "unsigned-integer-overflow",
             "signed-integer-overflow",
         ],
-        cfi: true,
     },
 }
 
@@ -107,7 +105,6 @@ cc_defaults {
     name: "libcodec2_soft_sanitize_cfi-defaults",
 
     sanitize: {
-        cfi: true,
         config: {
             cfi_assembly_support: true,
         },
diff --git a/media/codec2/sfplugin/utils/Android.bp b/media/codec2/sfplugin/utils/Android.bp
index 74e7ef1de0..edacaeb9d3 100644
--- a/media/codec2/sfplugin/utils/Android.bp
+++ b/media/codec2/sfplugin/utils/Android.bp
@@ -42,7 +42,6 @@ cc_library {
     ],
 
     sanitize: {
-        cfi: true,
         misc_undefined: [
             "unsigned-integer-overflow",
             "signed-integer-overflow",
diff --git a/media/libstagefright/bqhelper/Android.bp b/media/libstagefright/bqhelper/Android.bp
index e4fc128770..81bafbcb88 100644
--- a/media/libstagefright/bqhelper/Android.bp
+++ b/media/libstagefright/bqhelper/Android.bp
@@ -61,7 +61,6 @@ cc_defaults {
             "signed-integer-overflow",
             "unsigned-integer-overflow",
         ],
-        cfi: true,
     },
 }
 
diff --git a/media/libstagefright/flac/dec/Android.bp b/media/libstagefright/flac/dec/Android.bp
index 225c930dd1..4b48cc552d 100644
--- a/media/libstagefright/flac/dec/Android.bp
+++ b/media/libstagefright/flac/dec/Android.bp
@@ -36,7 +36,6 @@ cc_library {
             "signed-integer-overflow",
             "unsigned-integer-overflow",
         ],
-        cfi: true,
     },
 
     shared_libs: [
diff --git a/media/libstagefright/foundation/Android.bp b/media/libstagefright/foundation/Android.bp
index 407f609c29..27cb234265 100644
--- a/media/libstagefright/foundation/Android.bp
+++ b/media/libstagefright/foundation/Android.bp
@@ -117,7 +117,6 @@ cc_defaults {
             "unsigned-integer-overflow",
             "signed-integer-overflow",
         ],
-        cfi: true,
     },
 }
 
-- 
2.30.2

